src/Microsoft.AspNet.OData.Shared/Builder/LinkGenerationHelpers.cs

```diff
@@ -81,7 +82,16 @@ public static Uri GenerateSelfLink(this ResourceContext resourceContext, bool in
                 throw Error.Argument("resourceContext", SRResources.UrlHelperNull, typeof(ResourceContext).Name);
             }
 
-            IList<ODataPathSegment> navigationPathSegments = resourceContext.GenerateBaseODataPathSegments();
+            IList<ODataPathSegment> navigationPathSegments;
+            if (resourceContext.NavigationSource is IEdmContainedEntitySet &&
+                resourceContext.NavigationSource != resourceContext.SerializerContext.Path.NavigationSource)
```

Comment:
Do the serializer context and resource context have the right path information at this point? Because I think that they should not unless they are not set properly.

Reply:
@ElizabethOkerio They do. The idea behind having a test where the contained navigation property is on the path (e.g. `http://localhost/Menus(1)/Tabs`) and non-contained navigation property on the path (e.g. `http://localhost/Menus(1)/Panels`) was premised on the context having the right path information and that expectation held.